Source/Plugins/AutomapMode/AutomapMode.cs

```diff
@@ -454,29 +559,41 @@ public override void OnMouseLeave(EventArgs e)
 			base.OnMouseLeave(e);
 
 			// Highlight nothing
-			Highlight(null);
+			HighlightLine(null);
+			HighlightSector(null);
 		}
 
+		// Keyboard input handling; toggles a couple of options
+
 		public override void OnKeyDown(KeyEventArgs e)
 		{
 			base.OnKeyDown(e);
 
-			if(e.Control)
-			{
-				UpdateValidLinedefs();
-				General.Interface.RedrawDisplay();
-			}
+			UpdateOptions();
 		}
 
 		public override void OnKeyUp(KeyEventArgs e)
 		{
 			base.OnKeyUp(e);
 
-			if(!e.Control)
+			UpdateOptions();
+		}
+
+		private void UpdateOptions()
+		{
+			if(invertLineVisibility != General.Interface.CtrlState)
 			{
+				invertLineVisibility = General.Interface.CtrlState;
 				UpdateValidLinedefs();
 				General.Interface.RedrawDisplay();
 			}
+			if(editSectors != General.Interface.ShiftState)
+			{
+				editSectors = General.Interface.ShiftState;
+				HighlightLine(null);
+				HighlightSector(null);
+				General.Interface.RedrawDisplay();
+			}
 		}

```

Comment:
I noticed that pressing `Shift` only updates the sector highlight when th mouse is moved or the cursor is at a linedef's highlighting range. Not sure why it behaves like that, probably some interaction with other, unrelated, parts of the codebase, but it's a bit annoying in this usecase.